src/libcollections/string.rs

```diff
@@ -386,12 +418,14 @@ impl String {
     ///
     /// # Examples
     ///
+    /// Basic usage:
+    ///
     /// ```
     /// // 𝄞music
     /// let v = &[0xD834, 0xDD1E, 0x006d, 0x0075,
     ///           0x0073, 0x0069, 0x0063];
-    /// assert_eq!(String::from_utf16(v).unwrap(),
-    ///            "𝄞music".to_string());
+    /// assert_eq!(Some(String::from("𝄞music")),
+    ///            String::from_utf16(v));
```

Comment:
How come these assertions were swapped? I think it's generally `assert(actual, expected)` in terms of error messages at least.

I also think that `String::from` may be a bit distracting here when compared to `to_string`?